ssi-ldp/src/proof.rs

```diff
@@ -55,10 +57,20 @@ pub struct Proof {
 }
 
 impl Proof {
-    pub fn new(type_: &str) -> Self {
+    pub fn new(type_: ProofSuiteType) -> Self {
         Self {
-            type_: type_.to_string(),
-            ..Self::default()
+            type_,
```

Comment:
I was on the fence, I felt like deriving `Default` didn't make sense because depending on the features enabled you'd need a different default type, and in general it doesn't make sense to pick a type. But then because it's derived it would be available to end users and potentially introduce issues for them